frontend/src/main/scala/bloop/bsp/BspServer.scala

```diff
@@ -0,0 +1,79 @@
+package bloop.bsp
+
+import java.net.ServerSocket
+import java.util.Locale
+import java.nio.file.{Files, Path}
+
+import bloop.cli.{BspProtocol, Commands}
+import bloop.engine.{ExecutionContext, State}
+import com.martiansoftware.nailgun.{NGUnixDomainServerSocket, NGWin32NamedPipeServerSocket}
+import monix.execution.{Cancelable, Scheduler}
+
+object BspServer {
+  private[bloop] val isWindows: Boolean =
+    System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows")
+
+  import java.net.InetSocketAddress
+  private sealed trait ConnectionHandle { def socket: ServerSocket }
+  private case class WindowsLocal(pipeName: String, socket: ServerSocket) extends ConnectionHandle
+  private case class UnixLocal(path: Path, socket: ServerSocket) extends ConnectionHandle
+  private case class Tcp(address: InetSocketAddress, socket: ServerSocket) extends ConnectionHandle
```

Comment:
Why do we need `address`? How's that different from `socket.getInetAddress`?

Reply:
Good catch bro, `socket.getInetAddress` is better.